diff --git a/CMakeLists.txt b/CMakeLists.txt
index e90fe6c5..9bc57263 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,7 +40,7 @@ include(setupOptions)
 # some of our options are incompatible with e.g. GC
 if(USE_GC)
    set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-   add_subdirectory(libs/gc/gc-7.2f)
+   add_subdirectory(libs/gc/gc-8.0.2)
 endif()
 
 # 
diff --git a/libs/gc/gc-8.0.2/CMakeLists.txt b/libs/gc/gc-8.0.2/CMakeLists.txt
index d04abe95..61c4f4fc 100644
--- a/libs/gc/gc-8.0.2/CMakeLists.txt
+++ b/libs/gc/gc-8.0.2/CMakeLists.txt
@@ -259,3 +259,10 @@ IF(WIN32)
 ENDIF(WIN32)
 
 ADD_SUBDIRECTORY(tests)
+
+# TG: The static libmadx.a archive doesn't contain GC symbols needed to link
+# for 3rdparty applications. Therefore, ship libgc-lib.a  alongside libmadx.a:
+IF (NOT BUILD_SHARED_LIBS)
+  install(TARGETS gc-lib
+          ARCHIVE DESTINATION lib)
+ENDIF ()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 3a2b7e9b..91776cb6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -110,7 +110,7 @@ target_link_libraries(madx  ${CMAKE_THREAD_LIBS_INIT} ${STDCPP_LIBS})
 
 # Garbage collector:
 if(USE_GC)
-   include_directories(${CMAKE_SOURCE_DIR}/libs/gc/gc-7.2f/include)
+   include_directories(${CMAKE_SOURCE_DIR}/libs/gc/gc-8.0.2/include)
 endif()
 
 # Defining specific stuff for packaging:
