--- a/cmake/compilers/setupGNU.cmake
+++ b/cmake/compilers/setupGNU.cmake
@@ -6,8 +6,7 @@
 ###
 
 if (CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
-    # General:
-    set(CMAKE_Fortran_FLAGS " -fno-range-check -fno-f2c -cpp ") # remove -g -O2 from main list
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fno-range-check -fno-f2c -cpp ")
     execute_process(COMMAND ${CMAKE_Fortran_COMPILER} --version OUTPUT_VARIABLE CMAKE_Fortran_COMPILER_VERSION)
     string(REGEX MATCH "1?[0-9].[0-9].[0-9]" CMAKE_Fortran_COMPILER_VERSION ${CMAKE_Fortran_COMPILER_VERSION})
     if(${CMAKE_Fortran_COMPILER_VERSION} VERSION_GREATER 4.3.9)
